### ISSUE TITLE
Unclosed client session/unclosed connector on 3.0.1 shutdown or restart

### ISSUE THREAD

**Issue by aikaterna** [Cog-Creators/Red-DiscordBot#2474]
# Command bugs

#### Command name

restart or shutdown

#### What cog is this command from?

I don't believe this is being caused by a cog, unless it's core, as I would get this error with all cogs unloaded

#### What were you expecting to happen?

Restart or shutdown shuts down the bot processes cleanly

#### What actually happened?

Received this in the console:
```
_GatheringFuture exception was never retrieved
future: <_GatheringFuture finished exception=CancelledError()>
concurrent.futures._base.CancelledError
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x101e6e400>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x101fac8e8>, 7.72160103)]']
connector: <aiohttp.connector.TCPConnector object at 0x101e6e3c8>
```

#### How can we reproduce this issue?

Use Red 3.0.1, and use a shutdown or restart command.

While pip installing the bot on one machine I also noticed a version conflict with discord.py and aiohttp and websockets, but I did not save that info and I'm having trouble reproducing that. Also while trying to diagnose this I rolled back aiohttp to 3.4 on two machines but was still getting this error on shutdown.

**Comment by Tobotimus** [comment]
With regards to the version conflict with discord.py, we no longer depend on discord.py since it's vendored. Simply do `pip uninstall discord.py`

**Comment by aikaterna** [comment]
I don't have an alternate version of discord.py on this machine I am on right now (one of the ones that I've gotten that error on), so it is not uninstallable. pip3 freeze returns:
```
aiohttp==3.5.4
aiohttp-json-rpc==0.12
appdirs==1.4.3
async-timeout==3.0.1
attrs==18.2.0
black==18.9b0
chardet==3.0.4
Click==7.0
colorama==0.4.1
cycler==0.10.0
fuzzywuzzy==0.17.0
idna==2.8
idna-ssl==1.1.0
kiwisolver==1.0.1
markdown-to-json==1.0.0
matplotlib==3.0.2
multidict==4.5.2
numpy==1.16.1
Pillow==5.4.1
pyparsing==2.3.1
python-dateutil==2.8.0
python-Levenshtein==0.12.0
python-Levenshtein-wheels==0.13.1
PyYAML==3.13
raven==6.10.0
raven-aiohttp==0.7.0
Red-DiscordBot==3.0.1
Red-Lavalink==0.2.3
schema==0.6.8
six==1.12.0
toml==0.10.0
websockets==7.0
wordcloud==1.5.0
yarl==1.3.0
```

**Comment by Tobotimus** [comment]
That's likely an issue with your pip environment then, some old discord.py metadata might have hung around somewhere causing the version conflict message. If you have a virtual environment, I would nuke it and start again. If you're using system python, then bad luck :man_shrugging: you might be able to find some old discord.py `dist-info` or `egg-info` folder somewhere and delete it.

Essentially because the `discord` package is vendored within the *Red-DiscordBot* distribution, *discord.py* is a completely foreign concept to any Red installation, so any messages from pip saying anything about *discord.py* is probably due to old installations or dirty metadata.

**Comment by Tobotimus** [comment]
There's a bunch of stuff we need to fix up with regards to shutdown/cleanup. So I think this issue makes a good tracker for that.

**Comment by aikaterna** [comment]
Updating uvloop to 0.13.0 had resolved this for me previously, but I found that it returned on one of my test bots using dev recently. I created a couple test environments today on newly created vpses using Ubuntu 18.04.3 x64, installed both 3.1.6 and dev and received the same message on shutdown, no cogs loaded:
```
_GatheringFuture exception was never retrieved
future: <_GatheringFuture finished exception=CancelledError()>
concurrent.futures._base.CancelledError
```
edit: I did update both bot installs to uvloop 0.13.0 when testing this.

**Comment by Jackenmen** [comment]
I can confirm the same happening on current dev version on Windows with no cogs loaded as well.

**Comment by Drapersniper** [comment]
Got hub needs a me too button, no cogs loaded on dev and I see it on windows

**Comment by mikeshardmind** [comment]
This should be resolved as of #3261
